fiftyone/factory/repos/delegated_operation.py

```diff
@@ -310,13 +320,17 @@ def update_run_state(
             update["$set"]["status"] = progress
             update["$set"]["status"]["updated_at"] = datetime.utcnow()
 
+        collection_filter = {"_id": _id}
+        if required_state is not None:
+            collection_filter["run_state"] = required_state
+
         doc = self._collection.find_one_and_update(
-            filter={"_id": _id},
+            filter=collection_filter,
```

Comment:
I am not sure why we are passing it as a list. But I just tested it and both forms work.
I think I'd prefer if we not wrap in a list, because that feels more standard.